Quantum-Control-Applications/Superconducting/Single-Flux-Tunable-Transmon/17_cryoscope_1ns.py

```diff
@@ -270,18 +276,25 @@ def baked_waveform(waveform, pulse_duration):
             # Convert the results into Volts
             I, Q = u.demod2volts(I, readout_len), u.demod2volts(Q, readout_len)
             # Bloch vector Sx + iSy
-            qubit_state = (state[:, 0] * 2 - 1) + 1j * (state[:, 1] * 2 - 1)
+            Sx = (state[:, 0] * 2 - 1) - np.mean(
+                (state[:, 0] * 2 - 1)[zeros_before_pulse + 10 : len(flux_waveform) - zeros_after_pulse - 10]
+            )
+            Sy = (state[:, 1] * 2 - 1) - np.mean(
+                (state[:, 1] * 2 - 1)[zeros_before_pulse + 10 : len(flux_waveform) - zeros_after_pulse - 10]
+            )
+            qubit_state = Sx + 1j * Sy
```

Comment:
Why removing the average here?
Also the average will depend on the number of oscillation you get in Sx and Sy so one needs to be careful here...